scripts/o3de/o3de/repo_properties.py

```diff
@@ -17,13 +17,142 @@
 import logging
 import hashlib
 import shutil
+import requests
+import sys
+import msvcrt
+import time
+import re
 from packaging.version import Version, InvalidVersion
 from packaging.specifiers import SpecifierSet
 from o3de import manifest, utils, validation
 
 logger = logging.getLogger('o3de.repo_properties')
 logging.basicConfig(format=utils.LOG_FORMAT)
 
+def hide_credential(prompt):
```

Comment:
if we use `getpass` and add error handling if the input is empty so we can provide a hint to the user "to paste your key into a terminal use mouse right click" then we won't need this function.